### PR TITLE
spring data jpa

### PR DESCRIPTION
**Pull request by Leanchers**

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstore/model/Book.java [hunk 1]
```diff
@@ -9,10 +9,14 @@
...
 import lombok.Getter;
 import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Getter
 @Setter
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/model/Book.java]:
```suggestion
@SQLRestriction(valur = "is_deleted=false")
```

File: src/main/java/mate/academy/onlinebookstore/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -38,4 +38,27 @@ public BookDto findById(Long id) {
         );
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        Book updatedBook = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book by id " + id)
+        );
+        updatedBook.setTitle(requestDto.getTitle());
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/service/impl/BookServiceImpl.java]: Looks like Mapper logic, create method in Mapper to update book

File: src/main/java/mate/academy/onlinebookstore/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -38,4 +38,27 @@ public BookDto findById(Long id) {
...
+        updatedBook.setAuthor(requestDto.getAuthor());
+        updatedBook.setIsbn(requestDto.getIsbn());
+        updatedBook.setPrice(requestDto.getPrice());
+        updatedBook.setDescription(requestDto.getDescription());
+        updatedBook.setCoverImage(requestDto.getCoverImage());
+        return bookMapper.toDto(bookRepository.save(updatedBook));
+    }
+
+    @Override
+    public void deleteById(Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/service/impl/BookServiceImpl.java:52]:
```suggestion
    public void deleteById(Long id) {
         bookRepository.deleteById(id);
    }
```

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml:48]: Add empty line for each changelog

File: src/main/resources/liquibase.properties [hunk 5]
```diff
@@ -0,0 +1,4 @@
+url=jdbc:mysql://localhost:3306/book_store
+username=root
+password=1234
+changeLogFile=/db/changelog/db.changelog-master.yaml
```
Comment (Elena-Bruyako) [inline src/main/resources/liquibase.properties:4]: add empty line
```suggestion
changeLogFile=/db/changelog/db.changelog-master.yaml

```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml (hunk 4 above)
Comment (Leanchers) [reply src/main/resources/db/changelog/changes/01-create-books-table.yaml:48]: interesting, but I have an empty line in IDE

**Review by MAMentorFX — APPROVED** [review]

**Pull request merged**